pkgs/development/python-modules/quandl/default.nix

```
@@ -1,27 +1,20 @@
 {
-  lib, fetchFromGitHub, buildPythonPackage, isPy3k,
+  lib, fetchPypi, buildPythonPackage, isPy3k,
   # runtime dependencies
   pandas, numpy, requests, inflection, python-dateutil, six, more-itertools,
   # test suite dependencies
-  nose, unittest2, flake8, httpretty, mock, factory_boy, jsondate,
+  nose, unittest2, flake8, httpretty, mock, factory_boy, jsondate, parameterized, faker,
   # additional runtime dependencies are required on Python 2.x
   pyOpenSSL ? null, ndg-httpsclient ? null, pyasn1 ? null
 }:
 
 buildPythonPackage rec {
-  pname = "quandl";
-  version = "3.2.1";
-  sha256 = "0vc0pzs2px9yaqkqcmd2m1b2bq1iils8fs0xbl0989hjq791a4jr";
+  pname = "Quandl";
```

```suggestion
  pname = "quandl";
```
I think we're normalizing these. @FRidh?

---

Might be important for `fetchPypi` still.

---

This is why I changed the `pname`. Exactly because it had the wrong name in the first place. https://pypi.org/project/Quandl/

---

The attribute names and folders should be normalized but not the pname.

---

Shouldn't we just follow https://nixos.org/nixpkgs/manual/#sec-package-naming as normally?
> Don’t use uppercase letters in the name attribute — e.g., "mplayer-1.0rc2" instead of "MPlayer-1.0rc2".

---

Issue is that `pname` is used by fetchPypi to get the correct tarball

---

Then just set `pname = "Quandl";` in `fetchPypi`?

---

Sorry, @worldofpeace is diverging this conversation to triviality. Carry on.

---

@Mic92 and @FRidh what are your thoughts? I will use whatever your recommendations are including @worldofpeace.

---

`fetchPypi` failed because of the lowercase? Because on PyPI's side it seems https://pypi.org/project/quandl/ redirects to https://pypi.org/project/Quandl/ so it should not break the fetching.
Nix's naming guidelines suggest to use only lowercase, but if we have exceptions  for Python pkgs, Ruby pkgs, ... then the rule won't make much sense for contributors.

---

Seeing this tells me that what I'm suggesting is something that's being done https://github.com/costrouc/nixpkgs-dev-tools/commit/1953ac72be9cd4fdc8ee31be16c7999812c91a15 or should.

---

@worldofpeace you are right. I will set `pname` in `fetchPypi`